lib/puppet/provider/zabbix_hostgroup/ruby.rb

```diff
@@ -2,14 +2,32 @@
 Puppet::Type.type(:zabbix_hostgroup).provide(:ruby, parent: Puppet::Provider::Zabbix) do
   confine feature: :zabbixapi
 
+  def self.instances
+    api_hostgroups = zbx.hostgroups.all
+    api_hostgroups.map do |group_name, _id|
+      new(
+        ensure: :present,
+        name: group_name
+      )
+    end
+  end
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if (resource = resources[prov.name])
+        resource.provider = prov
+      end
+    end
+  end
+
   def create
     # Connect to zabbix api
     hgid = zbx.hostgroups.create(name: @resource[:name])
     hgid
```

Comment:
Is there any reason for assigning to `hgid` here?  `create` doesn’t actually need to return anything, does it?